simphony/testing/abc_check_engine.py

```diff
@@ -25,6 +30,11 @@ def setUp(self):
         self.items = self.create_dataset_items()
 
     @abc.abstractmethod
+    def container_factory(self, name, mode='w'):
```

Comment:
This is the method to get the engine to be tested, right?  Wouldn't this be called something like "container_engine" and there would be no need for parameters such as `name` or `mode`.

I understand that as there is a overlap with the H5CUDS in that it stores the DataSets like an ABCModelingEngine that these tests are used there too.